src/vector3.h

```diff
@@ -32,116 +32,123 @@ class alignas(sizeof(T)) vector3 {
+	inline friend vector3 operator*(const T scalar, const vector3 &a) { return a * scalar; }
 	//friend vector3 operator*(const double scalar, const vector3 &a) { return a*scalar; }
-	friend vector3 operator/(const vector3 &a, const float scalar)
+	inline friend vector3 operator/(const vector3 &a, const float scalar)
 	{
 		const T inv = 1.0 / scalar;
 		return vector3(a.x * inv, a.y * inv, a.z * inv);
 	}
-	friend vector3 operator/(const vector3 &a, const double scalar)
+	inline friend vector3 operator/(const vector3 &a, const double scalar)
 	{
 		const T inv = 1.0 / scalar;
 		return vector3(a.x * inv, a.y * inv, a.z * inv);
 	}
-	friend vector3 operator/(const T scalar, const vector3 &a)
+	inline friend vector3 operator/(const T scalar, const vector3 &a)
 	{
 		return vector3(scalar / a.x, scalar / a.y, scalar / a.z);
 	}
 
 	// component-wise ALL-less-than
-	auto operator<(const vector3 &b) const { return x < b.x && y < b.y && z < b.z; }
+	inline auto operator<(const vector3 &b) const { return x < b.x && y < b.y && z < b.z; }
 	// component-wise ALL-less-equal
-	auto operator<=(const vector3 &b) const { return x <= b.x && y <= b.y && z <= b.z; }
+	inline auto operator<=(const vector3 &b) const { return x <= b.x && y <= b.y && z <= b.z; }
 	// component-wise ALL-greater-than
-	auto operator>(const vector3 &b) const { return x > b.x && y > b.y && z > b.z; }
+	inline auto operator>(const vector3 &b) const { return x > b.x && y > b.y && z > b.z; }
 	// component-wise ALL-greater-equal
-	auto operator>=(const vector3 &b) const { return x >= b.x && y >= b.y && z >= b.z; }
+	inline auto operator>=(const vector3 &b) const { return x >= b.x && y >= b.y && z >= b.z; }
 
-	vector3 Cross(const vector3 &b) const { return vector3(y * b.z - z * b.y, z * b.x - x * b.z, x * b.y - y * b.x); }
-	T Dot(const vector3 &b) const { return x * b.x + y * b.y + z * b.z; }
-	T Length() const { return sqrt(x * x + y * y + z * z); }
-	T LengthSqr() const { return x * x + y * y + z * z; }
-	vector3 Lerp(const vector3 &b, const double percent) const
+	inline vector3 Cross(const vector3 &b) const { return vector3(y * b.z - z * b.y, z * b.x - x * b.z, x * b.y - y * b.x); }
+	inline T Dot(const vector3 &b) const { return x * b.x + y * b.y + z * b.z; }
+	inline T Length() const { return sqrt(x * x + y * y + z * z); }
+	inline T LengthSqr() const { return x * x + y * y + z * z; }
+	inline vector3 Lerp(const vector3 &b, const double percent) const
 	{
 		return *this + percent * (b - *this);
 	}
-	vector3 Normalized() const
+	inline void Normalize()
 	{
-		const T l = 1.0f / sqrt(x * x + y * y + z * z);
+		const T l = 1.0f / Length();
+		x *= l;
+		y *= l;
+		z *= l;
+	}
+	inline vector3 Normalized() const
+	{
+		const T l = 1.0f / Length();
 		return vector3(x * l, y * l, z * l);
 	}
-	vector3 NormalizedSafe() const
+	inline vector3 NormalizedSafe() const
 	{
-		const T lenSqr = x * x + y * y + z * z;
+		const T lenSqr = LengthSqr();
```

Comment:
Ditto - IMO `x*x + y*y + z*z` is not sufficiently complicated that having a single-source-of-truth is better than having optimized code in debug mode.